File: backend/GymRecommender/models/request.py
from django.db import models
from .account import Account

class Request(models.Model):
    id = models.UUIDField(primary_key=True)
    requested_at = models.DateTimeField()
    origin_latitude = models.FloatField()
    origin_longitude = models.FloatField()
    time_priority = models.IntegerField()
    total_cost_priority = models.IntegerField()
    min_congestion_rating = models.DecimalField(max_digits=4, decimal_places=2)
    min_rating = models.DecimalField(max_digits=4, decimal_places=2)
    min_membership_price = models.IntegerField()
    name = models.CharField(max_length=50, blank=True, null=True)
    user = models.ForeignKey(Account, models.DO_NOTHING)

    class Meta:
        managed = False
        db_table = 'request'
        unique_together = (('user', 'name'), ('user', 'requested_at'),)

File: backend/Models/Request.cs
﻿using System;
using System.Collections.Generic;
using backend.Enums;

namespace backend.Models;

public partial class Request
{
    public Guid Id { get; set; }

    public DateTime RequestedAt { get; set; }

    public double OriginLatitude { get; set; }

    public double OriginLongitude { get; set; }

    public int TimePriority { get; set; }

    public int TotalCostPriority { get; set; }

    public decimal MinCongestionRating { get; set; }

    public decimal MinRating { get; set; }

    public int MinMembershipPrice { get; set; }

    public string? Name { get; set; }
    
    public MembershipLength MembType { get; set; }

    public Guid UserId { get; set; }
    
    public TimeOnly? DepartureTime { get; set; }
    
    public TimeOnly? ArrivalTime { get; set; }

    public virtual ICollection<Recommendation> Recommendations { get; set; } = new List<Recommendation>();

    public virtual Account User { get; set; } = null!;
}
